scipy/stats/stats.py

```diff
@@ -8122,6 +8122,12 @@ def rankdata(a, method='average', *, axis=None):
     axis : {None, int}, optional
         Axis along which to perform the ranking. If ``None``, the data array
         is first flattened.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
```

Comment:
I'd suggest trying to keep this as consistent as possible with other `nan_policy` documentation.

In gh-13166, @WarrenWeckesser wrote:

```
        Defines how to handle the occurrence of nans in `a`.
        'propagate' returns nan, 'raise' raises an exception, 'omit'
        performs the calculations ignoring nan values. Default is
        'propagate'. Note that when the value is 'omit', nans in `scores`
        also propagate to the output, but they do not affect the ranks
        computed for the non-nan values.
```
So maybe adapt minimally to this function.

Reply:
OK. I will fix it. Do you think 'propagate' returns nan?  This might break backward compatibility. The [doc](http://scipy.github.io/devdocs/dev/api-dev/nan_policy.html) recommends that "this means just execute the function without checking for nan". Do you think which is better?

Reply:
I think you're probably right. Maybe you should make sure "propagate" does not change the behavior when you add this nan_policy, and that's why it should be the default. In this case, it will not return NaNs, because it didn't in the past.